### PR TITLE
Rename "positive electrode diffusivity" to "positive particle diffusivity"

### PR DESCRIPTION
**Pull request by julian-evers**
# Description

The "electrode diffusivity" parameter have been renamed to "particle diffusivity". parameter_values["Positive electrode diffusivity"] returns "Positive particle diffusivity" with a deprecation warning.

Fixes https://github.com/pybamm-team/PyBaMM/issues/3229

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by adityaraute — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 ## Features
 
+- Renamed "electrode diffusivity" to "particle diffusivity" as a non-breaking change with a deprecation warning ([#3624](https://github.com/pybamm-team/PyBaMM/pull/3624))
```
Comment (brosaplanella) [inline CHANGELOG.md:12]: Can you move this to the unreleased section?

**Review by brosaplanella — APPROVED** [review]
Looks good to me, just a small comment on the CHANGELOG. You should probably add a line in the breaking changes too.

File: CHANGELOG.md (hunk 1 above)
Comment (julian-evers) [reply CHANGELOG.md:22]: Sure, thank you for reviewing.

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +14,7 @@
 
 ## Breaking changes
 
+- Renamed "electrode diffusivity" to "particle diffusivity" as a non-breaking change with a deprecation warning ([#3624](https://github.com/pybamm-team/PyBaMM/pull/3624))
```
Comment (kratman) [inline CHANGELOG.md:17]: This says it is a non-breaking change, but it is listed under breaking changes

**Review by kratman — CHANGES_REQUESTED** [review]
Duplicate change log entry

File: CHANGELOG.md (hunk 2 above)
Comment (julian-evers) [reply CHANGELOG.md]: Thank you for reviewing. (I just deleted the line.)

**Pull request merged**